### PR TITLE
Add new unreliable_los option

### PR DESCRIPTION
**Pull request by aida-shumburo**
##### SUMMARY
Set interface configuration option for when to generate a 'loss of service' (LOS) alarm

##### GitHub Issues
"N/A"

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
sonic_interfaces

##### Ansible Report:
[add_unreliable_los.pdf](https://github.com/user-attachments/files/17154568/add_unreliable_los.pdf)

##### Raw Regression Test Output:
[unreliable_los.zip](https://github.com/user-attachments/files/17154960/unreliable_los.zip)

##### Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility or have provided any relevant "breaking_changes" descriptions in a "fragment" file in the "changelogs/fragments" directory of this repository.
- [x] I have provided a summary for this PR in valid "fragment" file format in the "changelogs/fragments" directory of this repository branch. Reference : [Ansible Change Log Document](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)

##### How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] I ran different ansible playbooks to test that the new ' unreliable_los' option value is being merged, replaced, deleted, and overriden the correct way
- [x] Ran a regression test (attached above)

### CODE REVIEW

File: tests/regression/roles/sonic_interfaces/defaults/main.yml [hunk 1]
```diff
@@ -41,18 +42,22 @@ tests:
         description: ansible Ethernet96
         mtu: 6444
         enabled: true
+        unreliable_los: UNRELIABLE_LOS_MODE_AUTO
```
Comment (kerry-meyer) [inline tests/regression/roles/sonic_interfaces/defaults/main.yml:45]: Change this to one of the other values to verify that changing the value works correctly.

File: tests/regression/roles/sonic_interfaces/defaults/main.yml [hunk 2]
```diff
@@ -196,6 +201,7 @@ tests:
         advertised_speed:
           - 100000
           - 40000
+        unreliable_los: UNRELIABLE_LOS_MODE_ON
```
Comment (kerry-meyer) [inline tests/regression/roles/sonic_interfaces/defaults/main.yml:204]: Please move this to test case 17 to make sure it is correctly deleted by this test (due to not being included in the "replacement" configuration for "Interface7").

File: tests/regression/roles/sonic_interfaces/defaults/main.yml
Comment (kerry-meyer) [inline tests/regression/roles/sonic_interfaces/defaults/main.yml:1]: - Please add a test case (line) that sets UNRELIABLE_LOS_MODE_OFF on an interface.
- Please also add deletion of one or more of the configured non-default unreliable_los values in test_case_06 ("Delete interface parameters") to verify correct handling of this operation.

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
Thank you for providing this new functionality.

I am posting a few minor comments to address before merging of this PR.

File: plugins/modules/sonic_interfaces.py [hunk 3]
```diff
@@ -107,6 +107,9 @@
         - FEC_DISABLED
         - FEC_DEFAULT
         - FEC_AUTO
+      unreliable_los:
```
[the remark this answers is not in the document]
Comment (aida-shumburo) [reply plugins/modules/sonic_interfaces.py:110]: I have documented the 'choices' below the description and type field.

File: plugins/module_utils/network/sonic/argspec/interfaces/interfaces.py [hunk 4]
```diff
@@ -65,7 +65,8 @@ def __init__(self, **kwargs):
                                     "FEC_FC",
                                     "FEC_DISABLED",
                                     "FEC_DEFAULT",
-                                    "FEC_AUTO"]}
+                                    "FEC_AUTO"]},
+                "unreliable_los": {"type": "str"}
```
[the remark this answers is not in the document]
Comment (aida-shumburo) [reply plugins/module_utils/network/sonic/argspec/interfaces/interfaces.py]: I have documented the allowable 'choices'.

File: tests/regression/roles/sonic_interfaces/defaults/main.yml (hunk 1 above)
Comment (aida-shumburo) [reply tests/regression/roles/sonic_interfaces/defaults/main.yml]: I have changed the value.

File: plugins/modules/sonic_interfaces.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply plugins/modules/sonic_interfaces.py:110]: Thank you.

Resolved.

File: plugins/modules/sonic_interfaces.py [hunk 5]
```diff
@@ -107,6 +107,9 @@
...
+        description: specifying when to generate a loss of service alarm.
```
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply plugins/modules/sonic_interfaces.py]: Thank you.

Resolved.

File: plugins/module_utils/network/sonic/argspec/interfaces/interfaces.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply plugins/module_utils/network/sonic/argspec/interfaces/interfaces.py]: Thank you.

Resolved.

File: tests/regression/roles/sonic_interfaces/defaults/main.yml [hunk 6]
```diff
@@ -82,9 +88,11 @@ tests:
         description:
       - name: "{{ interface8 }}"
         mtu: 6000
+        unreliable_los: UNRELIABLE_LOS_MODE_OFF
       - name: "{{ interface10 }}"
         enabled:
       - name: "{{ interface9 }}"
+        unreliable_los: UNRELIABLE_LOS_MODE_ON
```
Comment (kerry-meyer) [inline tests/regression/roles/sonic_interfaces/defaults/main.yml:95]: To test this deletion, line 60 will need to set UNRELIABLE_LOS_MODE_ON for this interface.

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
Almost all of the pending "Issues" are "Resolved" by your most recent set of changes.

I have one remaining "Issue" related to setting up the required conditions for integration test_06.

Everything else looks good.

File: tests/regression/roles/sonic_interfaces/defaults/main.yml (hunk 6 above)
Comment (kerry-meyer) [reply tests/regression/roles/sonic_interfaces/defaults/main.yml:95]: Thanks for making this change.

Resolved.

**Review by kerry-meyer — APPROVED** [review]
Thanks for making the final requested change.

All proposed code changes, added test cases, and test results look good.

Approved.

**Pull request merged**